### PR TITLE
REST high-level client: add get index API

### PR DESCRIPTION
**Pull request by sohaibiftikhar**
Relates to #27205

This is a followup to #31675.

- [x] Requires #31675 to be merged first.

### CODE REVIEW

**Comment by cbuescher** [comment]
@elasticmachine add to whitelist

File: client/rest-high-level/src/main/java/org/elasticsearch/client/IndicesClient.java [hunk 1]
```diff
@@ -437,6 +438,34 @@ public void getSettingsAsync(GetSettingsRequest getSettingsRequest, RequestOptio
...
+    /**
+     * Retrieve information about one or more indexes
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-index.html">
+     * Indices Get Index API on elastic.co</a>
+     * @param getIndexRequest the request
+     * @param options the request options (e.g. headers), use {@link RequestOptions#DEFAULT} if nothing needs to be customized
+     * @return the response
+     * @throws IOException in case there is a problem sending the request or parsing back the response
+     */
+    public GetIndexResponse getIndex(GetIndexRequest getIndexRequest, RequestOptions options) throws IOException {
```
Comment (javanna) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/IndicesClient.java]: can you call it just `get` please ?

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java [hunk 2]
```diff
@@ -834,6 +834,22 @@ static Request getSettings(GetSettingsRequest getSettingsRequest) {
...
+
+        String endpoint = endpoint(indices);
+        Request request = new Request(HttpGet.METHOD_NAME, endpoint);
+
+        Params params = new Params(request);
+        params.withIndicesOptions(getIndexRequest.indicesOptions());
+        params.withLocal(getIndexRequest.local());
+        params.withIncludeDefaults(getIndexRequest.includeDefaults());
+        params.withHuman(getIndexRequest.humanReadable());
+        params.withMasterTimeout(getIndexRequest.masterNodeTimeout());
```
Comment (javanna) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java]: can you also add this parameter to the corresponding REST spec? We are missing it there I think.

Comment (sohaibiftikhar) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java]: I don't understand what you mean by the REST spec? You mean the documentation for `get_index`?

Comment (javanna) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java]: https://github.com/elastic/elasticsearch/tree/master/rest-api-spec/src/main/resources/rest-api-spec/api , specifically https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/indices.get.json

Comment (sohaibiftikhar) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java]: Thanks! Now I know what this mysterious "spec" is every time you mention it 😀

Comment (javanna) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java]: 👍 haha I thought I had explained that, sorry for the confusion.

**Review by nik9000 — APPROVED** [review]
LGTM. @javanna, do you want to have another look?

**Review by javanna — APPROVED** [review]
LGTM thanks @nik9000 & @sohaibiftikhar !

**Pull request merged**